packages/sample-app/src/components/common/LoadPluginModal.tsx

```diff
@@ -0,0 +1,85 @@
+import { usePluginStore } from '@openshift/dynamic-plugin-sdk';
+import { Button, Modal, Text, TextContent, TextInput } from '@patternfly/react-core';
+import * as React from 'react';
+import { isValidURL } from '../../utils';
+
+const isValidPluginBaseURL = (value: string) => isValidURL(value) && value.endsWith('/');
+
+type LoadPluginModalProps = {
+  defaultBaseURL?: string;
+};
+
+export type LoadPluginModalRefProps = {
+  open: VoidFunction;
+};
+
+const LoadPluginModal = React.forwardRef<LoadPluginModalRefProps, LoadPluginModalProps>(
+  ({ defaultBaseURL = 'http://localhost:9001/' }, ref) => {
+    const [isModalOpen, setModalOpen] = React.useState(false);
+    const [baseURL, setBaseURL] = React.useState(defaultBaseURL);
+    const [baseURLValid, setBaseURLValid] = React.useState(isValidPluginBaseURL(defaultBaseURL));
+
+    const pluginStore = usePluginStore();
+
+    const onBaseURLChange = React.useCallback(
+      (value: string) => {
+        setBaseURL(value);
+        setBaseURLValid(isValidPluginBaseURL(value));
+      },
+      [setBaseURL, setBaseURLValid],
+    );
+
+    const openModal = React.useCallback(() => {
+      setModalOpen(true);
+    }, [setModalOpen]);
+
+    const closeModal = React.useCallback(() => {
+      setModalOpen(false);
+    }, [setModalOpen]);
+
+    const loadPlugin = React.useCallback(() => {
+      pluginStore.loadPlugin(baseURL);
+      closeModal();
+    }, [baseURL, pluginStore, closeModal]);
```

Comment:
@karelhala If you do `onClose={() => setModalOpen(false)}` you are re-creating the function upon each render.

This is typically not an issue when your component doesn't have children, but when it does, it will cause their re-render which may not be necessary. This may slow down rendering of complex component trees.

That said, I'll try to simplify the code as per your suggestion.